onnxruntime/core/framework/eager.cc

```diff
@@ -0,0 +1,147 @@
+#include <unordered_map>
+
+namespace onnxruntime {
+namespace eager {
+
+onnxruntime::Status CreateEagerOperator(const OrtKernelInfo* info,
+                                        const char* op_name,
+                                        const char* domain,
+                                        int version,
+                                        const char** type_constraint_names,
+                                        const ONNXTensorElementDataType* type_constraint_values,
+                                        int type_constraint_count,
+                                        const void* attr_values,
+                                        int attr_count,
+                                        OrtEagerOperator* op) {
+  *op = nullptr;
+  auto kernel_info = reinterpret_cast<const OpKernelInfo*>(info);
+  auto ep = reinterpret_cast<const IExecutionProvider*>(kernel_info->GetExecutionProvider());
+  auto kernel_registry = ep->GetKernelRegistry();
+  const KernelCreateInfo* kernel_create_info{};
+  std::unordered_map<std::string, MLDataType> type_constraint_map;
+  for (int i = 0; i < type_constraint_count; ++i) {
+    ONNX_NAMESPACE::TypeProto proto;
+    proto.mutable_tensor_type()->set_elem_type(type_constraint_values[i]);
+    type_constraint_map[type_constraint_names[i]] = DataTypeImpl::TypeFromProto(proto);
+  }
+  auto status = kernel_registry->TryFindKernel(op_name,
+                                               domain,
+                                               version,
+                                               type_constraint_map,
+                                               ep->Type(),
+                                               &kernel_create_info);
+  if (kernel_create_info == nullptr) {
+    return status;
+  }
+  onnxruntime::Node node;
+  auto onnx_attrs = reinterpret_cast<const ONNX_NAMESPACE::AttributeProto*>(attr_values);
+  for (int i = 0; i < attr_count; ++i) {
+    node.AddAttribute(onnx_attrs[i].name(), onnx_attrs[i]);
+  }
+  OpKernelInfo eagar_kernel_info(node, KernelDef{}, *ep, {}, {}, {});
+  std::unique_ptr<onnxruntime::OpKernel> op_kernel;
+  FuncManager func_mgr;
+  status = kernel_create_info->kernel_create_func(func_mgr, eagar_kernel_info, op_kernel);
+  if (status.IsOK()) {
+    *op = op_kernel.release();
+  }
+  return status;
+}
+
+onnxruntime::Status InvokeEagerOperator(const OrtKernelContext* context,
+                                        const OrtEagerOperator ort_op,
+                                        const OrtValue* const* input_values,
+                                        int input_count,
+                                        OrtValue* const* output_values,
+                                        int output_count) {
+  auto ctx = reinterpret_cast<const OpKernelContext*>(context);
+  AllocatorPtr allocator{};
+  auto ret = ctx->GetTempSpaceAllocator(&allocator);
+  if (!ret.IsOK()) {
+    return ret;
+  }
+  EagerKernelContext eager_ctx(input_values,
```

Comment:
Similar minor concern: can we either get rid of this struct, or inline/store it in the `OrtEagerOperator` (like in #4453)? These stack allocations seem unnecessary and could add a bit of overhead.

Reply:
Yes - it's better to save the overhead whenever possible, but we have to feed the kernel a "context" for computation, luckily by this PR, the overhead is just copying a few pointers. 
I read into the 4453, the design is elegant since it wraps up each kernel with a simple execution frame, right?
The thing is, even so, there has to be some overhead setting up inputs and outputs.